main_viewer.py

```diff
@@ -13,33 +20,52 @@
 
 import amp
 
+from typing import Dict, Any
+
 
 class MainViewer(QtWidgets.QMainWindow):
 
     def __init__(self):
+        # start typedef
+        self.statusbar: QtWidgets.QStatusBar
+        self.menuPlugins: QtWidgets.QMenu
+        self.menuEdit: QtWidgets.QMenu
+        self.menuFile: QtWidgets.QMenu
+        self.menubar: QtWidgets.QMenuBar
+        self.CohortTreeWidget: CohortTreeWidget
+        self.label_2: QtWidgets.QLabel
+        self.PlotListWidget: PlotListWidget
+        self.label: QtWidgets.QLabel
+        self.MplWidget: MplWidget
+        self.centralwidget: QtWidgets.QWidget
+        # end typedef
+
         super().__init__()
         # load ui elements into MainViewer class
         uic.loadUi("MainViewer.ui", self)
 
         # TODO: load cached plugins
-        self.plugins = {}
+        self.plugins: Dict[str, QtWidgets.QMainWindow] = {}
 
         # General UI threadpool (probably shouldn't use this for big algos)
         self.executer = concurrent.futures.ThreadPoolExecutor(max_workers=1)
 
         # connect UI callbacks
-        self.PlotListWidget.itemChanged.connect(self.onPlotItemChange)
-        self.PlotListWidget.currentItemChanged.connect(self.onPlotListChange)
-        self.CohortTreeWidget.itemClicked.connect(self.onFileToggle)
-        self.actionOpen_Cohort.triggered.connect(self.loadCohort)
-        self.actionAdd_Plugins.triggered.connect(self.addPlugin)
+        # TODO: find a good way to automatically determine the types for these
```

Comment:
Do we need to open an issue for this TODO?

Reply:
`parse_ui.py` solves this TODO and I forgot to remove it, my 🅱️